system/Model.php

```diff
@@ -679,18 +680,27 @@ public function update($id, $data)
 	//--------------------------------------------------------------------
 
 	/**
-	 * Deletes a single record from $this->table where $id matches
+	 * Deletes a single or many records from $this->table where $id matches
 	 * the table's primaryKey
 	 *
-	 * @param mixed $id    The rows primary key
+	 * deletes multiple items by a where clause
+	 * $model->where(array)->delete()
+	 *
+	 * deletes multiple items by ids
+	 * $model->delete(ids)
+	 *
+	 * deletes a single item by id
+	 * $model->delete(id)
+	 *
+	 * @param mixed $id    The rows primary key (values)
 	 * @param bool  $purge Allows overriding the soft deletes setting.
 	 *
 	 * @return mixed
 	 * @throws \CodeIgniter\Database\Exceptions\DatabaseException
 	 */
-	public function delete($id, $purge = false)
+	public function delete($id = null, $purge = false)
 	{
-		$this->trigger('beforeDelete', ['id' => $id, 'purge' => $purge]);
+		if ($id !== null && !is_array($id)) $id = [$id];
```

Comment:
Why to remove the `beforeDelete` trigger?

Reply:
That is not intentional, and it should be left in there.